packages/suite/src/actions/wallet/signVerifyActions.ts

```diff
@@ -1,133 +1,130 @@
 import TrezorConnect from 'trezor-connect';
-import { validateAddress } from '@wallet-utils/ethUtils';
 import * as notificationActions from '@suite-actions/notificationActions';
-import { SIGN_VERIFY } from './constants';
-import { Dispatch, GetState } from '@suite-types';
+import { GetState, Dispatch } from '@suite-types';
+import { copyToClipboard } from '@suite-utils/dom';
 
-export type inputNameType =
-    | 'signAddress'
-    | 'signMessage'
-    | 'signSignature'
-    | 'verifyAddress'
-    | 'verifyMessage'
-    | 'verifySignature';
+export const copyAddress = (address: string) => (dispatch: Dispatch) => {
+    const errorResponse = copyToClipboard(address, null);
 
-export type SignVerifyActions =
-    | { type: typeof SIGN_VERIFY.SIGN_SUCCESS; signSignature: string }
-    | { type: typeof SIGN_VERIFY.CLEAR_SIGN }
-    | { type: typeof SIGN_VERIFY.CLEAR_VERIFY }
-    | { type: typeof SIGN_VERIFY.INPUT_CHANGE; inputName: inputNameType; value: string }
-    | { type: typeof SIGN_VERIFY.TOUCH; inputName: inputNameType }
-    | { type: typeof SIGN_VERIFY.ERROR; inputName: inputNameType; message?: string };
-
-export const sign = (path: [number], message: string, hex = false) => async (
-    dispatch: Dispatch,
-    getState: GetState,
-) => {
-    const selectedDevice = getState().suite.device;
-    if (!selectedDevice) return;
-
-    const response = await TrezorConnect.ethereumSignMessage({
-        device: {
-            path: selectedDevice.path,
-            instance: selectedDevice.instance,
-            state: selectedDevice.state,
-        },
-        path,
-        hex,
-        message,
-        useEmptyPassphrase: selectedDevice.useEmptyPassphrase,
-    });
-
-    if (response && response.success) {
-        dispatch({
-            type: SIGN_VERIFY.SIGN_SUCCESS,
-            signSignature: response.payload.signature,
-        });
-    } else {
+    if (typeof errorResponse === 'string') {
         dispatch(
             notificationActions.addToast({
-                type: 'sign-message-error',
-                error: response.payload.error,
+                type: 'error',
+                error: errorResponse,
             }),
         );
+    } else {
+        dispatch(notificationActions.addToast({ type: 'copy-to-clipboard' }));
     }
 };
 
-export const verify = (address: string, message: string, signature: string, hex = false) => async (
+export const sign = (message: string, path: string, hex = false) => async (
     dispatch: Dispatch,
     getState: GetState,
-) => {
-    const selectedDevice = getState().suite.device;
-    if (!selectedDevice) return;
-    const error = validateAddress(address);
+): Promise<string | boolean> => {
```

Comment:
you don't have to type it, ts should do it automatically

nitpick: you dont have to return boolean since you're checking if response is a `typeof string`. you can just `return` (void)

Reply:
Yes, you can write this, and make from typescript -> javaScript, anyway I rewrote it